api/src/org/apache/cloudstack/api/command/admin/config/UpdateCfgCmd.java

```diff
@@ -117,6 +123,12 @@ public long getEntityOwnerId() {
 
     @Override
     public void execute() {
+        if (Strings.isNullOrEmpty(getCfgName())) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Empty configuration name provided");
+        }
+        if (getCfgName().equalsIgnoreCase(RoleService.EnableDynamicApiChecker.key())) {
+            throw new ServerApiException(ApiErrorCode.METHOD_NOT_ALLOWED, "Restricted configuration update not allowed");
```

Comment:
Method Not Allowed (405) indicates the caller attempted to use an HTTP method (i.e. GET, PUT, POST, DELETE) that is not allowed for the resource.  In this case, the method is allowed, but the parameters do not all the method to be performed.  Seems like ``PARAM_ERROR`` would be more appropriate.